aptos-move/framework/src/module_metadata.rs

```diff
@@ -241,7 +244,9 @@ pub fn check_metadata_format(module: &CompiledModule) -> Result<(), MalformedErr
                 bcs::from_bytes::<RuntimeModuleMetadataV1>(&data.value)
                     .map_err(|e| MalformedError::DeserializedError(data.key.clone(), e))?;
             }
-        } else if data.key == *COMPILATION_METADATA_KEY {
+        } else if features.is_enabled(FeatureFlag::REJECT_UNSTABLE_BYTECODE)
+            && data.key == *COMPILATION_METADATA_KEY
+        {
             if compilation_key_exist {
                 return Err(MalformedError::DuplicateKey);
```

Comment:
This is the wrong error, duplicate key?

Reply:
This follows the logic above at line 234 to check if there is a duplicated key. This check will be performed only when the feature `REJECT_UNSTABLE_BYTECODE` is enabled. If it is not enabled, UnknownKey will be raised instead (at the else branch below).